### PR TITLE
Fix #1965 - the hyperlink and the display name of the branch

### LINKED ISSUE #1965: The hyperlink to the branch is wrong if the branch is in a folder or the branch name has '#'
Suppose I have a branch named ```'feature/#1965'```, and the following history will be listed on the page of the dashboard feed after I push something to the repository.

***[john.doe](/john.doe) pushed to [#1965](/x_corp/project_x/src/#1965) at [x_corp/project_x](/x_corp/project_x)***

```html
<p class="text-bold">
    <a href="/john.doe">john.doe</a>
    pushed to <a href="/x_corp/project_x/src/#1965" rel="nofollow">#1965</a> at <a href="/x_corp/project_x" rel="nofollow">x_corp/project_x</a>
</p>
```

When I click the hyperlink, ```/x_corp/project_x/src/#1965```, the page show 'Not Found'.

The hyperlink of the branch, ```'feature/#1965'```, shall be:

 /x_corp/project_x/src/***feature/%231965***

and the branch name to display shall be ```'feature/#1965'```, not ```'#1965'```

### PR DESCRIPTION
**Pull request by arthuroy**
The hyperlink and the display name of the branch if the branch is in a folder or the branch name has '#'

I didn't commit modules/bindata/bindata.go, you need to re-generate and commit it.

Fix #1965

**Bug Example**

I setup a branch like the issue said at https://try.gogs.io/john.doe/project_x
And I had add your account to the collaborator, you can see what listed below on the dashboard afte you logined:

```
john.doe pushed to #1965 at john.doe/project_x

 ceb578ba4e Initial commit
 bfd5a61610 Initial commit
```

### CODE REVIEW

File: modules/git/utils.go [hunk 1]
```diff
@@ -35,6 +35,10 @@ func parsePrettyFormatLog(repo *Repository, logByts []byte) (*list.List, error)
 }
 
 func RefEndName(refStr string) string {
+	if strings.HasPrefix(refStr, "refs/heads/") {
+		return strings.TrimPrefix(refStr, "refs/heads/")
```
Comment (Unknwon) [inline modules/git/utils.go]: It would be better to use `return refStr[12:]`.

Comment (arthuroy) [inline modules/git/utils.go]: ok, or `refStr[len("refs/heads"):]`?
Maybe we shall have a const for "refs/heads"?

Comment (Unknwon) [inline modules/git/utils.go]: > ok, or `refStr[len("refs/heads"):]`?

Brilliant :joy:

**Comment by arthuroy** [comment]
@Unknwon I had committed modified modules/git/utils.go, please let me know if any problem.

**Comment by Unknwon** [comment]
OK. LGTM. Just to make sure, you have tested and confirm this PR fixes #1965 ?

**Comment by arthuroy** [comment]
@Unknwon 
Yes, I had tested on my local machine.

* You need to re-generate ***modules/bindata/bindata.go*** and commit it.
* Since the ***locale_en-US.ini*** is changed, UI using other locales will be broken.
* DB ***action.ref_name*** might needs to modify, e.g., ```#1965``` to ```feature/#1965```

**Comment by Unknwon** [comment]
OK, thanks!

> You need to re-generate modules/bindata/bindata.go and commit it.

Will do.

> Since the locale_en-US.ini is changed, UI using other locales will be broken.

I will fix them on Crowdin.

> DB action.ref_name might needs to modify, e.g., #1965 to feature/#1965

I guess you're the only one has this issue, otherwise was reported by someone else :trollface:

**Comment by arthuroy** [comment]
Thanks XD

**Pull request merged**